x-pack/plugins/observability_onboarding/public/hooks/use_install_system_integration.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import { useTrackedPromise } from '@kbn/use-tracked-promise';
+import { i18n } from '@kbn/i18n';
+import { useKibana } from './use_kibana';
+
+// Errors
+const UNAUTHORIZED_ERROR = i18n.translate(
+  'xpack.observability_onboarding.installSystemIntegration.error.unauthorized',
+  {
+    defaultMessage:
+      'Required kibana privilege {requiredKibanaPrivileges} is missing, please add the required privilege to the role of the authenticated user.',
+    values: {
+      requiredKibanaPrivileges: "['Fleet']",
+    },
+  }
+);
+
+type ErrorType = 'AuthorizationError' | 'UnknownError';
+export interface SystemIntegrationError {
+  type: ErrorType;
+  message: string;
+}
+
+type IntegrationInstallStatus =
+  | 'installed'
+  | 'installing'
+  | 'install_failed'
+  | 'not_installed';
+
+export const useInstallSystemIntegration = ({
+  onIntegrationCreationSuccess,
+  onIntegrationCreationFailure,
+}: {
+  onIntegrationCreationSuccess: ({ version }: { version?: string }) => void;
+  onIntegrationCreationFailure: (error: SystemIntegrationError) => void;
+}) => {
+  const {
+    services: { http },
+  } = useKibana();
+  const [createIntegrationRequest, callCreateIntegration] = useTrackedPromise(
```

Comment:
Perhaps keep the neutral naming. At least to me they are easier to understand (I don't have prior experience with `useTrackedPromise`)

```suggestion
  const [requestState, performRequest] = useTrackedPromise(
```

Reply:
Alternatively (quite verbose)

```suggestion
  const [systemIntegrationRequestState, performSystemIntegrationRequest] = useTrackedPromise(
```